ios/Plugin/Plugin.swift

```diff
@@ -76,4 +76,18 @@ public class FirebaseAnalytics: CAPPlugin {
             call.reject("name is missing")
         }
     }
+
+    @objc func enable(_ call: CAPPluginCall) {
+        DispatchQueue.main.async {
+            Analytics.setAnalyticsCollectionEnabled(true)
+            call.success()
+        }
+    }
+  
+    @objc func disable(_ call: CAPPluginCall) {
+            DispatchQueue.main.async {
+            Analytics.setAnalyticsCollectionEnabled(false)
+            call.success()
+        }
+    }
```

Comment:
Plugin calls already use a dispatch queue within Capacitor, AFAIK. Is this necessary?

Reply:
Can [confirm](https://github.com/ionic-team/capacitor/blob/cdd03370a5f7c9be97c3a7b45e3ca24e9ea95b44/ios/Capacitor/Capacitor/CAPBridge.swift#L425)  that dispatch queue is invoked with js call. What would be the reason for [built in plugins](https://github.com/ionic-team/capacitor/search?p=1&q=DispatchQueue.main.async&unscoped_q=DispatchQueue.main.async) to call dispatch queue separately?

Reply:
Took from my plugin, back on earlier days of capacitor it was necessary.

Reply:
Not sure if it still necessary, though. From what I remember it crashes the app if we not dispatch async.

Reply:
I've just tested it again. Safe to remove the dispatch queue.